moto/sns/models.py

```diff
@@ -388,7 +388,17 @@ def update_sms_attributes(self, attrs):
         self.sms_attributes.update(attrs)
 
     def create_topic(self, name, attributes=None, tags=None):
-        fails_constraints = not re.match(r"^[a-zA-Z0-9_-]{1,256}$", name)
+
+        if (
+            attributes.get("FifoTopic")
+            and attributes.get("FifoTopic").lower() == "true"
+        ):
+            fails_constraints = not re.match(
+                r"^[a-zA-Z0-9_.-]{1,256}$", name
```

Comment:
This regex would allow something like `name.something.fifo`, which is not valid.
A better regex would probably be `^[a-zA-Z0-9_-]{1,256}.fifo$`